### PR TITLE
feat(external tables): supports syncing external table files

### PR DESCRIPTION
**Pull request by zijiacj**
#### What type of PR is this?
type-feature

#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java [hunk 1]
```diff
@@ -183,6 +185,19 @@ public GenerateTableDDLResp generateUpdateDDL(@NotNull ConnectionSession session
                 .build();
     }
 
+    public SuccessResponse syncExternalTableFiles(@NotNull ConnectionSession connectionSession, String schemaName,
+            @NotBlank String externalTableName) {
+        DBSchemaAccessor schemaAccessor = DBSchemaAccessors.create(connectionSession);
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java:190]: may not use accessor directly in service layer, use extension instead please

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java [hunk 2]
```diff
@@ -183,6 +185,19 @@ public GenerateTableDDLResp generateUpdateDDL(@NotNull ConnectionSession session
                 .build();
     }
 
+    public SuccessResponse syncExternalTableFiles(@NotNull ConnectionSession connectionSession, String schemaName,
```
[the remark this answers is not in the document]
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java:188]: ok

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java (hunk 1 above)
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java:190]: ok

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java [hunk 3]
```diff
@@ -122,4 +123,14 @@ public ListResponse<DataType> getPartitionKeyDataTypes(@PathVariable String sess
...
 
+    @PutMapping(value = "/{sessionId}/databases/{databaseName}/externalTables/"
+            + "{externalTableName}/syncExternalTableFiles")
+    public SuccessResponse syncExternalTableFiles(@PathVariable String sessionId,
+            @PathVariable(required = false) String databaseName,
+            @PathVariable(required = true, name = "externalTableName") String externalTableName) {
+        Base64.Decoder decoder = Base64.getDecoder();
+        externalTableName = new String(decoder.decode(externalTableName));
+        ConnectionSession session = sessionService.nullSafeGet(sessionId, true);
+        return tableService.syncExternalTableFiles(session, databaseName, externalTableName);
```
[the remark this answers is not in the document]
Comment (zijiacj) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:134]: may stay the same with getTable()

File: server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBTableService.java:188]: return void

**Review by yhilmare — DISMISSED** [review]
lgtm

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java [hunk 4]
```diff
@@ -122,4 +123,15 @@ public ListResponse<DataType> getPartitionKeyDataTypes(@PathVariable String sess
         return Responses.list(this.partitionPlanService.getPartitionKeyDataTypes(sessionId, databaseId, tableName));
     }
 
+    @PutMapping(value = "/{sessionId}/databases/{databaseName}/externalTables/"
```
Comment (MarkPotato777) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:126]: Shoud it be `@PostMapping`?

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java [hunk 5]
```diff
@@ -122,4 +123,15 @@ public ListResponse<DataType> getPartitionKeyDataTypes(@PathVariable String sess
...
+            + "{externalTableName}/syncExternalTableFiles")
+    public SuccessResponse syncExternalTableFiles(@PathVariable String sessionId,
+            @PathVariable(required = false) String databaseName,
+            @PathVariable(required = true, name = "externalTableName") String externalTableName) {
+        Base64.Decoder decoder = Base64.getDecoder();
+        externalTableName = new String(decoder.decode(externalTableName));
+        ConnectionSession session = sessionService.nullSafeGet(sessionId, true);
+        tableService.syncExternalTableFiles(session, databaseName, externalTableName);
+        return Responses.empty();
```
Comment (MarkPotato777) [inline server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:135]: Why return `empty`?

Comment (zijiacj) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:135]: empty response indicates that synchronization is successful

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java (hunk 4 above)
Comment (zijiacj) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java:126]: `@PutMapping` stands for update, `@PostMapping` stands for new, I think `@PutMapping` is more appropriate

**Review by PeachThinking — APPROVED** [review]
lgtm

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java (hunk 4 above)
Comment (zijiacj) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java]: use @PostMapping

File: server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java (hunk 5 above)
Comment (zijiacj) [reply server/odc-server/src/main/java/com/oceanbase/odc/server/web/controller/v2/DBTableController.java]: use Boolean

**Review by yhilmare — APPROVED** [review]
lgtm

**Review by MarkPotato777 — APPROVED** [review]
LGTM

**Review by PeachThinking — APPROVED** [review]
lgtm

**Pull request merged**